src/Imports/ModelImporter.php

```diff
@@ -42,6 +42,7 @@ public function import(Worksheet $worksheet, ToModel $import, int $startRow = 1)
         $progessBar       = $import instanceof WithProgressBar;
         $withMapping      = $import instanceof WithMapping;
         $withCalcFormulas = $import instanceof WithCalculatedFormulas;
+        $rememberRow      = method_exists($import, 'setRowNumber');
```

Comment:
Is there a reason for using a trait instead of an interface like every other "feature"? Could be called "WithRowNumbers" then (just an idea).

Reply:
As the Trait only implements a basic setter and getter. This would be a good option as `istanceof` is less "expensive" than `method_exists`.

Reply:
Just like with the chunk offset trait, I prefer not to have a concern for this.

Reply:
I have not seen this comment and changed RemembersRowNumber to an Interface in my latest commit. As it is much simpler now I really think the Interface with only the setter for the current row makes more sense in this case.

Have a Look at it. If you still prefer the Trait I can change it to a Trait again.

Reply:
Still prefer keeping the trait instead of the interface for this. 
For me the concerns are used when the export/import needs to know something about the export/import. The traits are small helpers to make things easier, having an interface and a trait for a setter feels a bit too heavy for me.

Reply:
> Still prefer keeping the trait instead of the interface for this.

I will change the Code to Trait later today.

But still just want to dump some thoughts about Interface vs Trait:

> having an interface and a trait for a setter feels a bit too heavy for me.

Actually as it is right now we only have an Interface and no Trait(Name would have to changed from RemembersRowNumber to WithRowNumber or similar).
The only thing an Importer has to implement is the Method `setRowNumber(int $rowNumber);`.

> For me the concerns are used when the export/import needs to know something about the export/import. 

Thats true, the other Interfaces are there to provide Information from the implementation of an Importer towards running the importer. And I see the point in being consistent in that sense.


I have one more concern though. The Trait-Implementation could be considered as a breaking change as somebody could have implemented a `setRowNumber()` function inside their importer. The Trait-Implementation would not know that this is to be ignored.
The Interface Implementation can be considered free of Breaking changes as we check `instanceof` istead of `method_exists`.

Reply:
> Actually as it is right now we only have an Interface and no Trait(Name would have to changed from RemembersRowNumber to WithRowNumber or similar).

Still same feeling then :) Would have to implement the setRowNumber method myself, which adds lot of noise to the import class. I prefer just adding a trait to my import and be done with it :) I don't think it will be a very common use case for someone to want to do anything different than setting the row number in the class property.

>  have one more concern though. The Trait-Implementation could be considered as a breaking change as somebody could have implemented a setRowNumber() function inside their importer. The Trait-Implementation would not know that this is to be ignored.

We can also rename the `setRowNumber` to `rememberRowNumber()` that's probably more unique and less likely to clash.